### DISCUSSION TITLE
Does prisma/client conflict with react-refresh

### DISCUSSION THREAD

**Discussion by tmcw** [prisma/prisma#14939] [Q&A] [answered]
Hi folks, this is not quite a bug report because it's in between `react-refresh` and `@prisma/client` and my application, and I'm not sure which one is to blame. Here's the traceback:

```
Unhandled Runtime Error
TypeError: Invalid enum value: $$typeof

Call Stack
Object.get
node_modules/@prisma/client/runtime/index-browser.js (95:0)
Object.register
node_modules/react-refresh/cjs/react-refresh-runtime.development.js (357:0)
Object.registerExportsForReactRefresh
node_modules/@next/react-refresh-utils/internal/helpers.js (54:0)
eval
webpack-internal:///./db/index.ts (33:31)
./db/index.ts
file:///Users/tmcw/s/placemark/.next/static/chunks/pages/_app.js (302:1)
options.factory
/_next/static/chunks/webpack.js (685:31)
```

As far as I can tell, the issue here is that Prisma generates this object with getters:

https://github.com/prisma/prisma/blob/6d7f68d4b3c52e90c909dde95a6638799fb9417b/packages/client/src/runtime/strictEnum.ts#L30-L42

That will throw if you try to get an unknown property from it.

Then react-refresh, (in versions [before 2020](https://github.com/facebook/react/commit/dfb6a403357d5ac2c0ea87aa6e5a4f947de342f8)), gets a `.typeof` property:

https://github.com/facebook/react/blob/37cb732c59863297e48f69ac1f6e2ba1aa1886f0/packages/react-refresh/src/ReactFreshRuntime.js#L324-L333

And that property access throws. And - I know, I'm pinned to an old react-refresh release and will try updating that, but in the meantime - this is a bit surprising! l'm not sure, have others encountered this problem?

**Comment by austincrim** [comment, marked as answer]
Hey there @tmcw 👋,

This does look like a bug to me. Would you mind filling out a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with your relevant environment and version info? Our engineers will take it from there 👍.

**Comment by rmiller61** [comment]
I have also experienced this bug on v4.3. Has there been any update to this?

**Reply by austincrim** [reply]
If you can create a minimal reproduction repo, please share it on [this issue](https://github.com/prisma/prisma/issues/14953) so our engineers can take a look 👍.

**Comment by mfkrause** [comment]
Also experiencing the same error. I'm not using `react-refresh` standalone, but it is being required by Metro in a React Native / Expo project. The stacktrace leads from the above mentioned line in `@prisma/client` to `metro-runtime` v0.70.3, and then to the above mentioned line of `react-refresh`. In `metro-runtime`, the `require` polyfill seems to be affected from the stacktrace (`node_modules/metro-runtime/src/polyfills/require.js (904:23)`), which is line 918 in the source file [here](https://github.com/facebook/metro/blob/v0.70.3/packages/metro-runtime/src/polyfills/require.js).

My schedule currently doesn't allow me to spin up a min repro, but just wanted to throw this in. Also, I was able to temporarily bypass this by simply adding `$$typeof` to the `allowList` of the getter function so that it doesn't throw anymore but returns `void 0` instead. I didn't dig too deep into the code to understand what all this fuzz actually does, so please take this with a grain of salt. This is the patch I've applied:

```diff
diff --git a/node_modules/@prisma/client/runtime/index-browser.js b/node_modules/@prisma/client/runtime/index-browser.js
index a4a94bd..8274d51 100644
--- a/node_modules/@prisma/client/runtime/index-browser.js
+++ b/node_modules/@prisma/client/runtime/index-browser.js
@@ -78,6 +78,7 @@ var objectEnumValues = {
 var allowList = /* @__PURE__ */ new Set([
   "toJSON",
   "asymmetricMatch",
+  "$$typeof",
   Symbol.iterator,
   Symbol.toStringTag,
   Symbol.isConcatSpreadable,
diff --git a/node_modules/@prisma/client/runtime/index.js b/node_modules/@prisma/client/runtime/index.js
index 12ad838..4276d8d 100644
--- a/node_modules/@prisma/client/runtime/index.js
+++ b/node_modules/@prisma/client/runtime/index.js
@@ -32001,6 +32001,7 @@ __name(warnAboutRejectOnNotFound, "warnAboutRejectOnNotFound");
 // src/runtime/strictEnum.ts
 var allowList = /* @__PURE__ */ new Set([
   "toJSON",
+  "$$typeof",
   "asymmetricMatch",
   Symbol.iterator,
   Symbol.toStringTag,
```

**Comment by AndrewSouthpaw** [comment]
BTW this is fixed! Discovered this in the latest Prisma v5.5.2. 😄

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
